src/Calculator/Common/KeyboardShortcutManager.cpp

```diff
@@ -421,215 +365,159 @@ void KeyboardShortcutManager::OnCharacterReceivedHandler(CoreWindow ^ sender, Ch
     int viewId = Utils::GetWindowId();
     auto currentHonorShortcuts = s_fHonorShortcuts.find(viewId);
 
-    if (currentHonorShortcuts != s_fHonorShortcuts.end())
+    if (currentHonorShortcuts == s_fHonorShortcuts.end() || currentHonorShortcuts->second)
     {
-        if (currentHonorShortcuts->second)
-        {
-            wchar_t character = static_cast<wchar_t>(args->KeyCode);
-            auto buttons = s_CharacterForButtons.find(viewId)->second.equal_range(character);
-            RunFirstEnabledButtonCommand(buttons);
+        wchar_t character = static_cast<wchar_t>(args->KeyCode);
+        auto buttons = s_CharacterForButtons.find(viewId)->second.equal_range(character);
+        RunFirstEnabledButtonCommand(buttons);
 
-            LightUpButtons(buttons);
-        }
+        LightUpButtons(buttons);
     }
 }
 
-const std::multimap<MyVirtualKey, WeakReference>& GetCurrentKeyDictionary(MyVirtualKey key, bool altPressed = false)
+const multimap<MyVirtualKey, WeakReference>* KeyboardShortcutManager::GetCurrentKeyDictionary(bool altPressed)
 {
     int viewId = Utils::GetWindowId();
 
-    if (altPressed)
-    {
-        return s_VirtualKeyAltChordsForButtons.find(viewId)->second;
-    }
-    else if (
-        (s_ShiftKeyPressed.find(viewId)->second)
-        && ((Window::Current->CoreWindow->GetKeyState(VirtualKey::Control) & CoreVirtualKeyStates::Down) == CoreVirtualKeyStates::Down))
-    {
-        return s_VirtualKeyControlShiftChordsForButtons.find(viewId)->second;
-    }
-    else if (s_ShiftKeyPressed.find(viewId)->second)
+    bool isControlKeyPressed = (Window::Current->CoreWindow->GetKeyState(VirtualKey::Control) & CoreVirtualKeyStates::Down) == CoreVirtualKeyStates::Down;
```

Comment:
In OnKeyDownHandler in some cases we'll call CoreWindow->GetKeyState twice, and we could get into an unexpected state if CoreWindow doesn't return consistent results.

I think the caller should be responsible for passing in the state of all three modifier keys, not just alt.

Reply:
I'm fine with that!